src/main/java/net/sf/rails/ui/swing/StartRoundWindow.java

```diff
@@ -293,66 +285,58 @@ private void initCells() {
-                    gameUIManager.getORUIManager(),
-                    si);
-            addField(f, itemNameXOffset, itemNameYOffset + i, 1, 1, WIDE_RIGHT);
+            final StartItem si = round.getStartItem(i);
+
+            itemName[i] = new Caption(si.getId());
+            HexHighlightMouseListener.addMouseListener(itemName[i], gameUIManager.getORUIManager(), si);
+            addField(itemName[i], itemNameXOffset, itemNameYOffset + i, 1, 1, WIDE_RIGHT);
+
+            itemNameButton[i] = new ClickField(si.getId(), "", "", this, itemGroup);
+            HexHighlightMouseListener.addMouseListener(itemNameButton[i], gameUIManager.getORUIManager(), si);
+            addField(itemNameButton[i], itemNameXOffset, itemNameYOffset + i, 1, 1, WIDE_RIGHT);
+
             // Prevent row height resizing after every buy action
             itemName[i].setPreferredSize(itemNameButton[i].getPreferredSize());
 
             if (showBasePrices) {
-                f = basePrice[i] = new Field(si.getBasePriceModel());
-                addField(f, basePriceXOffset, basePriceYOffset + i, 1, 1, 0);
+                basePrice[i] = new Field(si.getBasePriceModel());
+                addField(basePrice[i], basePriceXOffset, basePriceYOffset + i, 1, 1, 0);
             }
 
             if (includeBidding) {
-                f = minBid[i] = new Field(round.getMinimumBidModel(i));
-                addField(f, minBidXOffset, minBidYOffset + i, 1, 1, WIDE_RIGHT);
+                minBid[i] = new Field(round.getMinimumBidModel(i));
+                addField(minBid[i], minBidXOffset, minBidYOffset + i, 1, 1, WIDE_RIGHT);
             }
 
             for (int j = 0; j < np; j++) {
-                f = bidPerPlayer[i][j] = new Field(round.getBidModel(i, players.getPlayerByPosition(j)));
-                addField(f, bidPerPlayerXOffset + j, bidPerPlayerYOffset + i,
-                        1, 1, 0);
+                bidPerPlayer[i][j] = new Field(round.getBidModel(i, players.getPlayerByPosition(j)));
+                addField(bidPerPlayer[i][j], bidPerPlayerXOffset + j, bidPerPlayerYOffset + i, 1, 1, 0);
             }
 
-            f = info[i] = new Field(infoIcon);
-            f.setToolTipText(getStartItemDescription(si));
-            HexHighlightMouseListener.addMouseListener(f,
-                    gameUIManager.getORUIManager(),
-                    si);
-            addField(f, infoXOffset, infoYOffset + i, 1, 1, WIDE_LEFT);
+            info[i] = new Field(infoIcon);
+            info[i].setToolTipText(getStartItemDescription(si));
+            HexHighlightMouseListener.addMouseListener(info[i], gameUIManager.getORUIManager(), si);
+            addField(info[i], infoXOffset, infoYOffset + i, 1, 1, WIDE_LEFT);
 
             // Invisible field, only used to hold current item status.
-            f = itemStatus[i] = new Field(si.getStatusModel());
+            itemStatus[i] = new Field(si.getStatusModel());
```

Comment:
What is meant here by "invisible"? As I see it this field is never added to the UI, therefore I propose removing it.

This field is currently used to store a value which is later parsed into an int. I think we should replace this usage with a direct call to `si.getStatusModel()`

Reply:
I think the `Field` name is misleading here-- while yes, it is a `JLabel`, it's actually implementing `Observer` -- key to note as it means whenever the underlying `IntegerState` is changed the value stored in the `JLabel` is changed, which is then retrieved in `setItemNameButton()`.

So creation of something like an int array and storing off the result won't work-- it won't be updated.

So either we store off the `IntegerState` returned from the `si.getStatusModel()` call or we create our own observer to watch over that `IntegerState`. 

I'm not sure keeping the reference to `si` is appropriate as it that is actually a change in logic-- we assume `si` is valid long term but it very well may not be.

Reply:
In the end this will most likely be replaced by an `IntegerBinding` in the UI class. If possible I would refrain from using bindings or observers/observables in the model/state classes long-term because they tend to be highly linked to UI implementations. 

Ideally I would like to change our UI to induce its own (UI) updates in an explicit way, but we are not there yet.